src/utils.js

```diff
@@ -165,3 +166,42 @@ export function ensureDefinedConfig(object, requester) {
     }
   });
 }
+
+/**
+ * This function generates and returns the API config url based on the following environment variables:
+ *
+ *  MFE_CONFIG_API_PATH: API config path, this must not include the domain.
+ *  MFE_CONFIG_API_PORT(Optional): Specify the port, useful in development environments
+ *  MFE_CONFIG_API_REGEX: This regex will be used to generate the config domain based on the current domain.
+ *  APP_ID(Optional): This value makes reference to specific settings for a MFE.
+ *
+ * @returns {string}
+ */
+export function generateAPIConfigUrl() {
+  const {
+    APP_ID,
+    MFE_CONFIG_API_PATH,
+    MFE_CONFIG_API_PORT,
+    MFE_CONFIG_API_REGEX,
+  } = getConfig();
+
+  if (!MFE_CONFIG_API_PATH || !MFE_CONFIG_API_REGEX) {
```

Comment:
thinking in carrying this to upstream...here I have some question related some approach may the community should use with MFEs.
It exist the possibility to serve the MFEs in the same domain of the LMS_HOST. Something like
LMS_HOST/learning . In this case the function would work without `MFE_CONFIG_API_REGEX`. 
And  in that case would match only the `window.location.hostname`.

https://github.com/mitodl/mitxonline/issues/42
https://github.com/mitodl/mitxonline/issues/42#issuecomment-897618718

Reply:
in that case the regex should be the same domain 

![image](https://user-images.githubusercontent.com/36200299/216457254-9a3715d3-af1d-4b6d-b58d-2fc9b24d8ca7.png)

Reply:
Sure but the window.location.hostname is variable(tenant), and `MFE_CONFIG_API_REGEX` not? 🤔🤔🤔.
So I have to set a `MFE_CONFIG_API_REGEX` that always match. And also this forces me to set it .

Reply:
yes, if you want to use this you have to set it otherwise you have the default option

Reply:
Knowing that if I want to use MFE multitenant with the form `LMS_HOST/learning`
I have to use the regex `MFE_CONFIG_API_REGEX= ".*"` if I want to work with the multitenat way. 

![image](https://user-images.githubusercontent.com/51926076/216466360-a84b3b34-543e-4160-8826-19a384f53fc1.png)

Reply:
the point is that the person, who wants to use this, has to find the regex that fits for their case.
for your previous example is better
![image](https://user-images.githubusercontent.com/36200299/216469212-5975ae1d-7289-43c3-9f8d-08973f1d9331.png)

since your regex matches with an empty space and generates a dot at the end